HTTP/HttpListenerApp/Program.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+
+class Program
+{
+    private const string URL = "http://localhost:8888/";
+    private static HttpListener? server;
+
+    static async Task Main(string[] args)
+    {
+        StartServer();
+
+        while (true)
+        {
+            await HandleIncomingRequest();
+        }
+    }
+
+    private static void StartServer()
+    {
+        server = new HttpListener();
+        server.Prefixes.Add(URL);
+        server.Start();
+
+        Console.WriteLine("Server started...");
+    }
+
+    private static async Task HandleIncomingRequest()
+    {
+        HttpListenerContext context = await server.GetContextAsync();
+        HttpListenerRequest request = context.Request;
+
+        string resourceName = request.RawUrl.Trim('/');
+
+        switch (resourceName)
+        {
+            case "MyName":
+                await SendNameInResponse(context.Response);
+                break;
+            case "Information":
+                SetResponseStatus(context.Response, 100);
+                break;
+            case "Success":
+                SetResponseStatus(context.Response, 200);
+                break;
+            case "Redirection":
+                SetResponseStatus(context.Response, 300);
+                break;
+            case "ClientError":
+                SetResponseStatus(context.Response, 400);
+                break;
+            case "ServerError":
+                SetResponseStatus(context.Response, 500);
+                break;
+            case "MyNameByHeader":
+                var response = context.Response;
+                response.Headers.Add("X-MyName", "Ceren");
+                response.StatusCode = 200;
+                response.Close();
+                break;
+            case "MyNameByCookies":
+                SetMyNameByCookies(context.Response);
+                break;
+            default:
+                SetResponseStatus(context.Response, 404);
+                break;
+        }
+
+        context.Response.Close();
+    }
+
+    private static void SetResponseStatus(HttpListenerResponse response, int statusCode)
+    {
+        response.StatusCode = statusCode;
+    }
+
+    private static async Task SendNameInResponse(HttpListenerResponse response)
+    {
+        string name = "Gerund";
+        byte[] buffer = System.Text.Encoding.UTF8.GetBytes(name);
+        response.ContentLength64 = buffer.Length;
+
+        await response.OutputStream.WriteAsync(buffer);
+    }
+
+    private static void SetMyNameByCookies(HttpListenerResponse response)
+    {
+        var cookie = new Cookie("MyName", "Cero from Cookie");
+        response.Cookies.Add(cookie);
+        response.StatusCode = 200;
```

Comment:
In general, in ASP.Net Core, strongly advise to use [StatusCodes](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.statuscodes?view=aspnetcore-7.0) class for setting HTTP status codes